tests/src/cli.js

```diff
@@ -4,82 +4,145 @@
 import path from 'path';
 
 import { expect } from 'chai';
-import { CLIEngine } from 'eslint';
+import { CLIEngine, ESLint } from 'eslint';
 import eslintPkg from 'eslint/package.json';
 import semver from 'semver';
+import * as importPlugin from '../../src/index';
 
 describe('CLI regression tests', function () {
   describe('issue #210', function () {
+    let eslint;
     let cli;
     before(function () {
-      cli = new CLIEngine({
-        useEslintrc: false,
-        configFile: './tests/files/issue210.config.js',
-        rulePaths: ['./src/rules'],
-        rules: {
-          'named': 2,
-        },
-      });
+      if(ESLint) {
+        eslint = new ESLint({
+          useEslintrc: false,
+          overrideConfigFile: './tests/files/issue210.config.js',
+          rulePaths: ['./src/rules'],
+          overrideConfig: {
+            rules: {
+              'named': 2,
+            },
+          },
+          plugins: { 'eslint-plugin-import': importPlugin },
+        });
+      } else {
+        cli = new CLIEngine({
+          useEslintrc: false,
+          configFile: './tests/files/issue210.config.js',
+          rulePaths: ['./src/rules'],
+          rules: {
+            'named': 2,
+          },
+        });
+        cli.addPlugin('eslint-plugin-import', importPlugin);
+      }
     });
     it("doesn't throw an error on gratuitous, erroneous self-reference", function () {
-      expect(() => cli.executeOnFiles(['./tests/files/issue210.js'])).not.to.throw();
+      if (eslint) {
+        return eslint.lintFiles(['./tests/files/issue210.js'])
+          .then(() => { }, () => expect.fail() );
+      } else {
+        expect(() => cli.executeOnFiles(['./tests/files/issue210.js'])).not.to.throw();
+      }
     });
   });
 
   describe('issue #1645', function () {
+    let eslint;
     let cli;
     beforeEach(function () {
       if (semver.satisfies(eslintPkg.version, '< 6')) {
         this.skip();
       } else {
-        cli = new CLIEngine({
-          useEslintrc: false,
-          configFile: './tests/files/just-json-files/.eslintrc.json',
-          rulePaths: ['./src/rules'],
-          ignore: false,
-        });
+        if (ESLint) {
+          this.timeout(15000); // In the ESLint v8 test, it takes a lot of time to generate the ESLint class because we linked eslint-plugin-import.
```

Comment:
It seems that 10 seconds is not enough, so I increased it to 15 seconds.

https://github.com/import-js/eslint-plugin-import/pull/2191#discussion_r689981103

Reply:
I investigated why link takes time.
It took a long time to try to put the coverage mark on "lib" by "nyc" because link no longer is node_modules.
I think it can be fixed by adding "lib" to "exclude" in ".nycrc".

Reply:
gotcha